src/loginAzurePowerShell.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+
+var psPath: string;
+
+export const initializeAz = async (servicePrincipalId: string, servicePrincipalKey: string, tenantId: string, subscriptionId: string) => {
+    psPath = await io.which("pwsh", true);
```

Comment:
Here we are doing explicit check for pwsh command. If pwsh doesn't exist, the action will throw and login will fail. For non-powershell based actions, it is not really required to login into Azure PowerShell. Logging into Azure CLI should be enough. We should check if we want to explicitly check for pwsh. 

Refer: https://github.com/actions/toolkit/blob/5859d7172e5f4281198f6d74eeb9fe96fb27a23d/packages/io/src/io.ts#L175

Reply:
as discussed offline, action will fail if powershell login fails because we do not have input to control it.